experimental/dds/tree2/src/core/tree/delta.ts

```diff
@@ -245,25 +274,100 @@ export interface MoveIn {
  */
 export interface Insert<TTree = ProtoNode> extends HasModifications<TTree> {
 	readonly type: typeof MarkType.Insert;
+
 	// TODO: use a single cursor with multiple nodes instead of array of cursors.
 	/**
 	 * Must be of length 1 when `fields` is populated.
 	 */
 	readonly content: readonly TTree[];
 
 	/**
-	 * When set, indicates that the inserted content should be deleted.
-	 * This is used in scenarios where content is moved out from under an inserted subtree that is then deleted.
+	 * The ID to assign the first node being replaced.
+	 * Subsequent replaced nodes should be assigned incrementing IDs.
+	 * Populated iff the insertion is transient.
+	 */
+	readonly detachId?: DetachedNodeId;
+
+	/**
+	 * When set, indicates that the inserted content is replacing some existing content.
+	 */
+	readonly oldContent?: {
+		/**
+		 * Modifications to the old content.
+		 */
+		readonly fields?: FieldMarks<TTree>;
+		/**
+		 * The ID to assign the first node being replaced.
+		 * Subsequent replaced nodes should be assigned incrementing IDs.
+		 */
+		readonly detachId: DetachedNodeId;
+	};
+}
+
+/**
+ * Describes the restoration of a contiguous range of node.
+ * @alpha
+ */
+export interface Restore<TTree = ProtoNode> {
+	readonly type: typeof MarkType.Restore;
+	/**
+	 * Must be 1 when `fields` is populated.
+	 */
+	readonly count: number;
+
+	/**
+	 * When set, indicates that the inserted content is replacing some existing content.
+	 */
+	readonly oldContent?: {
+		/**
+		 * Modifications to the old content.
+		 */
+		readonly fields?: FieldMarks<TTree>;
+
+		/**
+		 * The ID to assign the first node being replaced.
+		 * Subsequent replaced nodes should be assigned incrementing IDs.
+		 */
+		readonly detachId: DetachedNodeId;
+	};
```

Comment:
Since this is exactly the same as the type used in `Insert.oldContent`, it would be nice if they used a shared type definition. Maybe `Insert` and `Restore` should also share an interface with the `oldContent` field.